connection_params.ts

```diff
@@ -17,106 +18,117 @@ function getPgEnv(): IConnectionParams {
   }
 }
 
-function selectFrom(
-  sources: Array<IConnectionParams>,
-  key: keyof IConnectionParams,
-): string | undefined {
-  for (const source of sources) {
-    if (source[key]) {
-      return source[key];
-    }
-  }
-
-  return undefined;
+function isDefined<T>(value: T): value is Exclude<T, undefined | null> {
+  return value !== undefined && value !== null;
 }
```

Comment:
I think this could be simply:
```
function assert(val): asserts expr {}
```

Reply:
Do you mean just invoking the built in `assert` function in place of this?

This type guard function is used in 3 places, and only one of them could actually be replaced with an assertion function, in the other two places it is just used as a type guard.

When I did `assert(val !== null && val !== undefined)` for the assertion, typescript did not recognise that `val` would be non-nullable after that line. So I lost the type safety.

I did however replace the `Exclude<T, undefined | null>` return type with the built in type `NonNullable<T>`.

Reply:
> Do you mean just invoking the built in `assert` function in place of this?

There's no built-in `assert`. I guess this solution is fine as well.

Reply:
Yeah thanks, I don't know how I figured that. I had node typings loaded for some reason when I just tried it in the editor.